### ISSUE TITLE
[topgen] Remove duplicated license headers from autogen files

### ISSUE THREAD

**Issue by vogelpi** [lowRISC/opentitan#2929]
A couple of our auto-generated files have duplicated license headers. One copy is inserted by `util/topgen.py` and the other copy comes in via template file. In addition some auto-generated files do not contain the warning that they are auto-generated. 

We should probably:
1. Make sure all our templates have a license banner that does not get into the autogen file.
1. Change topgen to only insert the license banner together with a warning that this is auto-generated to all auto-generated files.

Currently, the following files have a duplicated banner:
```
grep -Rc --include \*.sv SPDX- * | grep ":2"
hw/top_earlgrey/ip/clkmgr/rtl/autogen/clkmgr.sv:2                                                                                                                                                                                    
hw/top_earlgrey/ip/clkmgr/rtl/autogen/clkmgr_pkg.sv:2                                                                                                                                                                                       
hw/top_earlgrey/ip/flash_ctrl/rtl/autogen/flash_ctrl.sv:2                                                                                                                                                                                   
hw/top_earlgrey/ip/flash_ctrl/rtl/autogen/flash_ctrl_region_cfg.sv:2                                                                                                                                                                        
hw/top_earlgrey/ip/flash_ctrl/rtl/autogen/flash_ctrl_pkg.sv:2
```

**Comment by tjaychen** [comment]
ooops! this is my bad.
I think it's probably a result of the ip level example not having that header (thus included in the template), and `topgen.py` having it.
I'll fix the clkmgr ones.

**Comment by vogelpi** [comment]
Hi @tjaychen,
no worries. I wanted to look into this at some point but will have to do some other stuff first. I also noted that some autogen files have a line saying something like "this file has been generated with the following command" and then there is no command. I thought to solve both these things in one shot.

**Comment by msfschaffner** [comment]
@vogelpi is this still pending or can we close this?

**Comment by vogelpi** [comment]
Yes, it's still an issue. For current master I got:
```
grep -Rc --include \*.sv SPDX- * | grep ":2"
hw/top_earlgrey/ip/flash_ctrl/rtl/autogen/flash_ctrl_region_cfg.sv:2
hw/top_earlgrey/ip/flash_ctrl/rtl/autogen/flash_ctrl.sv:2
hw/top_earlgrey/ip/flash_ctrl/rtl/autogen/flash_ctrl_pkg.sv:2
hw/top_earlgrey/ip/clkmgr/rtl/autogen/clkmgr_pkg.sv:2
hw/top_earlgrey/ip/clkmgr/rtl/autogen/clkmgr.sv:2
hw/top_earlgrey/ip/rstmgr/rtl/autogen/rstmgr_pkg.sv:2
hw/top_earlgrey/ip/rstmgr/rtl/autogen/rstmgr.sv:2
hw/top_earlgrey/ip/rstmgr/dv/sva/autogen/rstmgr_rst_en_track_sva_if.sv:2
```
However, this is really ultra-low priority. And I believe it would still be a good starter task.

**Comment by matutem** [comment]
These duplicate license headers don't get generated by the ipgen flow, so they will be gone once #8440 is complete. As it stands the ones in rstmgr are already gone.

**Comment by rswarbrick** [comment]
I ran a search like this again:
```
$ git grep -c SPDX | grep :2
hw/ip/entropy_src/rtl/entropy_src_main_sm_pkg.core:2
hw/ip/trial1/doc/trial1.html:2
hw/vendor/lowrisc_ibex/dv/uvm/core_ibex/directed_tests/gen_testlist.py:2
quality/BUILD.bazel:2
site/book-theme/css/chrome.css:2
site/book-theme/css/general.css:2
site/book-theme/css/variables.css:2
site/book-theme/index.hbs:2
sw/device/silicon_owner/tock/kernel/layout.ld:2
sw/device/silicon_owner/tock/upstream_kernel/layout.ld:2
sw/host/hsmtool/scripts/pkcs11_consts.py:2
sw/host/ot_certs/src/codegen.rs:2
util/autogen_banner.py:2
util/doc/vendor.md:2
util/ipgen/renderer.py:2
util/py/scripts/gh_testplan.py:2
util/reggen/gen_tock.py:2
util/reggen/ip_block.py:2
util/regtool.py:2
util/rom_chip_info.py:2
util/sh/lib/banners.sh:2
util/sh/scripts/bin2c.sh:2
util/topgen.py:2
util/vendor.py:2
```

I'm walking through the hits now.

**Comment by rswarbrick** [comment]
I think there are three types of genuine hits in the list:

- Two files from tock (now tracked in #27340). This isn't actually an example of what *this* issue is tracking, so I think we can leave addressing it to that issue.
- An example in `trial1`. The mangled license headers seem to have come from a manual import at the start of time, so it's not really related to this issue specifically.
- A core file in `entropy_src`. (A mistake in the initial handwritten version). Fixed by #27341.

But there aren't actually any other examples in autogen files, so I think this issue can be closed.
